app/controllers/contacts_controller.rb

```diff
@@ -1,15 +1,34 @@
 class ContactsController < ApplicationController
   before_action :set_contact, only: %i[edit update destroy]
-  before_action :set_contact_type_options, only: %i[create edit new update]
   before_action :set_new_contact_from_params, only: :create
-  before_action :set_contact_type, only: %i[update create]
+  before_action :set_contact_type, only: :create
 
-  def index
-    @contacts = Contact.includes([:contact_type]).order(:name).decorate
+  def new
+    @contact_types = CategoryReference.list_by_category(:contact_type)
+
+    if request.get?
+      @contact_type = ContactType.new
+      return false
+    end
+
+    @contact_type = ContactType.new(contact_type_params)
+
+    if @contact_type.valid?
+      redirect_to new_new_details_contacts_path
```

Comment:
When this redirect happens the params are being lost. You either need to add them to the URL, or submit this form to the new_details page.

Reply:
My preference would be to submit to the next page and if the form object is invalid, re-render the new page.